docs/TBG_macros.cfg

```diff
@@ -132,6 +132,26 @@ TBG_radiation="--<species>_radiation.period 1 --<species>_radiation.dump 2 --<sp
 #--<species>_transRad.period   Gives the number of time steps between which the radiation should be calculated.
 TBG_transRad="--<species>_transRad.period 1000"
 
+# The following flags are available for the saxs plugin.
+# For a full description, see the plugins section in the online documentation.
+#--<species>_transRad.period    Period at which the plugin is enabled.
+#--<species>_saxs.outputPeriod    Period at which the accumulated amplitude is written to the output file.
```

Comment:
I am not sure if the particle based plugin should be separate. I would prefer it to be the same plugin, with an option to use either density or particle based calculation.
So the naming as saxs would be good in that case.

Reply:
Okay - I think implementing two different kernels/interfaces on top of the same `saxx.hpp` file should be no problem but I might be mistaken.
@psychocoderHPC and @sbastrakov what do you think?

Reply:
This depends if the configuration and feature set is equal.
Never the less if the changes are in we should merge it as it is and if possible unify in a separate PR.

Reply:
Technically, there is no problem to have two different kernels and conditionally call one depending on parameters. I think it's more a matter of do we consider the density-based and particle-based ones as just two different ways of computing (or estimating) the same values, with the same parameters provided by the user, and same type of output? If so, I feel they would be better as the same plugin. However, in case there is a difference in input parameters and postprocessing, perhaps there would be not much use in merging the two.

Reply:
It is (or rather should be/will be) more or less the same feature set, calculating the same physical quantity (I should add that I see no reason one should ever use the particle approach in a PIC code, but that's another story). We also had the idea once to merge everything with Richard's radiation plugin!
And I also support merging as is, unifying later.